integrated_channels/integrated_channel/models.py

```diff
@@ -60,5 +78,133 @@ class EnterpriseCustomerPluginConfiguration(TimeStampedModel):
     )
     active = models.BooleanField(blank=False, null=False)
 
+    GRADE_PASSING = 'Pass'
+    GRADE_FAILING = 'Fail'
+    GRADE_UNKNOWN = 'In Progress'
+
     class Meta:
         abstract = True
+
+    @staticmethod
+    def channel_code():
+        """
+        Returns an capitalized identifier for this channel class, unique among subclasses.
+        """
+        raise NotImplementedError(_('Implemented in concrete subclass.'))
+
+    def transmit_learner_data(self):
+        """
+        Collect and transmit learner data for the ``EnterpriseCustomer``.
+        """
+        transmitter = self.get_learner_data_transmitter()
+        for learner_data in self.collect_learner_data():
+            transmitter.transmit(learner_data)
+
+    def collect_learner_data(self):
+        """
+        Collect learner data for the ``EnterpriseCustomer`` where data sharing consent is granted.
+
+        Yields a learner data object for each enrollment, containing:
+
+        * ``enterprise_enrollment``: ``EnterpriseCourseEnrollment`` object.
+        * ``certificate``: ``GeneratedCertificate`` for the user+course; None if not found.
+          "Course completion" occurs when course certificates are issued.
+        """
+
```

Comment:
At this point have we already made the determination that the EnterpriseCustomer has enabled data sharing consent?  If not then this might be a good spot to do that check.